core-bundle/src/Image/Studio/LightBoxResult.php

```diff
@@ -0,0 +1,196 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Image\Studio;
+
+use Contao\Image\PictureConfiguration;
+use Contao\LayoutModel;
+use Contao\PageModel;
+use Contao\StringUtil;
+use Psr\Container\ContainerInterface;
+use Webmozart\PathUtil\Path;
+
+class LightBoxResult
+{
+    /**
+     * @readonly
+     *
+     * @var ContainerInterface
+     */
+    protected $locator;
+
+    /**
+     * @readonly
+     *
+     * @var ImageResult|null
+     */
+    private $image;
+
+    /**
+     * @readonly
+     *
+     * @var string|null
+     */
+    private $uri;
+
+    /**
+     * @readonly
+     *
+     * @var string|null
+     */
+    private $groupIdentifier;
+
+    /**
+     * @param array|PictureConfiguration|int|string|null $sizeConfiguration
+     *
+     * @internal Use the `contao.image.studio` factory to get an instance of this class.
+     */
+    public function __construct(ContainerInterface $locator, string $uri, $sizeConfiguration = null, string $groupIdentifier = null)
+    {
+        $this->locator = $locator;
+        $this->uri = $uri;
+        $this->groupIdentifier = $groupIdentifier;
+
+        if ($this->isValidForResizing($uri)) {
+            $filePath = Path::isAbsolute($uri) ?
+                Path::canonicalize($uri) :
+                Path::makeAbsolute($uri, $this->projectDir());
+
+            $this->image = $locator
+                ->get('contao.image.studio')
+                ->createImage($filePath, $sizeConfiguration ?? $this->getDefaultLightBoxSizeConfiguration())
+            ;
+        }
+    }
+
+    /**
+     * Return if this light box result contains a (resized) image.
+     */
+    public function hasImage(): bool
+    {
+        return null !== $this->image;
+    }
+
+    /**
+     * Return the underlying (resized) image.
+     */
+    public function getImage(): ImageResult
+    {
+        if (!$this->hasImage()) {
+            throw new \RuntimeException('This light box result does not contain a (resized) image.');
+        }
+
+        return $this->image;
+    }
+
+    /**
+     * Return the link url pointing to the resource.
+     */
+    public function getLinkHref(): string
+    {
+        return $this->hasImage() ? $this->image->getImageSrc() : $this->uri;
+    }
+
+    /**
+     * Return the light box group identifier.
+     */
+    public function getGroupId(): string
+    {
+        return $this->groupIdentifier ?? $this->createFallbackGroupIdentifier();
+    }
+
+    private function projectDir(): string
+    {
+        return $this->locator
+            ->get('parameter_bag')
+            ->get('kernel.project_dir')
+            ;
+    }
+
+    /**
+     * todo: who is responsible for this? can we maybe move this to the ImageFactory?
+     */
+    private function isValidForResizing(string $uri): bool
```

Comment:
I’m not sure about this. Mixing relative/absolute URLs/paths could be problematic. E.g. relative URLs need to be URL-decoded. Can we split the uri into two constructor parameters uri and path?

Reply:
Agreed. This might make things more clear.

Reply:
I splitted the arguments + moved the logic out of the `LightBoxResult`. It's in the factory for now (only temporary).

I *think* it should probably go to the `FigureBuilder` but I wonder if people would need that all the time (because  typically you're picking an URL or a file in the backend within in the same field). :thinking: